huaweicloud/services/acceptance/dds/data_source_huaweicloud_dds_backups_test.go

```diff
@@ -0,0 +1,150 @@
+package dds
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDatasourceDdsBackups_basic(t *testing.T) {
+	dataSource := "data.huaweicloud_dds_backups.all"
+	dc := acceptance.InitDataSourceCheck(dataSource)
+	rName := acceptance.RandomAccResourceName()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceBackups_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(dataSource, "backups.#"),
+
+					resource.TestCheckOutput("is_backup_id_filter_useful", "true"),
+					resource.TestCheckOutput("is_backup_name_filter_useful", "true"),
+					resource.TestCheckOutput("is_backup_type_filter_useful", "true"),
+					resource.TestCheckOutput("is_instance_id_filter_useful", "true"),
+					resource.TestCheckOutput("is_instance_name_filter_useful", "true"),
+					resource.TestCheckOutput("is_status_filter_useful", "true"),
+					resource.TestCheckOutput("is_description_filter_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDatasourceBackups_basic(name string) string {
+	return fmt.Sprintf(`
+%s
+
+data "huaweicloud_dds_backups" "all" {
+  depends_on = [huaweicloud_dds_backup.test]
+}
+
+// filter by backup_id
+data "huaweicloud_dds_backups" "filter_by_backup_id" {
+  backup_id  = huaweicloud_dds_backup.test.id
```

Comment:
```suggestion
  backup_id = huaweicloud_dds_backup.test.id
```
and fix other places